core/src/main/java/com/linecorp/armeria/client/DefaultClientFactory.java

```diff
@@ -171,4 +175,22 @@ public void close() {
     void doClose() {
         clientFactoriesToClose.forEach(ClientFactory::close);
     }
+
+    private URI normalizeUri(URI uri, Scheme scheme) {
+        if (isUndefinedUri(uri)) {
+            // We use a special singleton marker URI for clients that do not explicitly define a
+            // host or scheme at construction time.
+            // As this isn't created by users, we don't need to normalize it.
+            return uri;
+        }
+
+        if (scheme.serializationFormat() != SerializationFormat.NONE) {
+            return uri;
+        }
+
+        final String newScheme = uri.getScheme()
+                                    .replace(SerializationFormat.NONE.uriText(), "")
+                                    .replace("+", "");
```

Comment:
Can we just replace it once because `nonehttp://...` was not valid uri?

Reply:
I couldn't replace it once, as scheme could be http+none or none+http. Hence I had to do twice.

I have updated the code, where there is no longer replace. We are using parsed Scheme.sessionProtocol() as prefix for new URI construction.